pkg/curl/parser.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package curl
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	flag "github.com/spf13/pflag"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+)
+
+func parseCommands(command Commands) (*CommandFlags, error) {
+	flagset := flag.NewFlagSet("curl", flag.ContinueOnError)
+	flagset.ParseErrorsWhitelist.UnknownFlags = true
+
+	// these parts of flags are referenced to the manual of curl
+	flagset.BoolP("silent", "s", true, "silent mode")
+	location := flagset.BoolP("location", "L", false, "follow the location")
+	requestMethod := flagset.StringP("request", "X", "GET", "request method")
+	rawHeader := flagset.StringArrayP("header", "H", []string{}, "HTTP extra header")
+	data := flagset.StringP("data", "d", "", "data")
+	err := flagset.Parse(command)
+
+	// first non-flag arg is the command itself, use the second non-flag arg as the url.
+	if flag.NArg() > 1 {
+		return nil, fmt.Errorf("can not find the url")
+	}
+	url := flagset.Arg(1)
+
+	if err != nil {
+		return nil, nil
+	}
```

Comment:
Should we add some logs here? And I think this error could be returned before `url:=flagset.Arg(1)`.

Reply:
Updated in 3105f41c, PTAL!